packages/components/src/toolbar/toolbar-context/index.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * External dependencies
+ */
+import type { ToolbarStateReturn } from 'reakit/Toolbar';
+
+/**
+ * WordPress dependencies
+ */
+import { createContext } from '@wordpress/element';
+
+const ToolbarContext = createContext< ToolbarStateReturn | undefined >(
+	undefined
+);
```

Comment:
`ToolbarStateReturn` was used as it is the inferred type for `toolbarState`.

Please refer to [ToolbarContainer](https://github.com/mike-day/gutenberg/blob/refactor/toolbar-context-to-ts/packages/components/src/toolbar/toolbar/toolbar-container.tsx#L27-L31).